File: lancersBackEnd/model/Contracts.py
from datetime import datetime
from enum import Enum

class Etat(Enum):
    Attente = 1
    EnCours = 2
    Fini = 3
    Retard = 4
    Rejected = 5
    Approved = 6

class Contracts:
    def __init__(self, id: int, email: str, idService: int, dateDebut: datetime, dateLivraison: datetime, prix: float, etat: Etat):
        self._id = id
        self._email = email
        self._idService = idService
        self._dateDebut = dateDebut
        self._dateLivraison = dateLivraison
        self._prix = prix
        self._etat = etat

    @property
    def id(self) -> int:
        return self._id

    @id.setter
    def id(self, value: int):
        self._id = value

    @property
    def email(self) -> str:
        return self._email

    @email.setter
    def email(self, value: str):
        self._email = value

    @property
    def idService(self) -> int:
        return self._idService

    @idService.setter
    def idService(self, value: int):
        self._idService = value

    @property
    def dateDebut(self) -> datetime:
        return self._dateDebut

    @dateDebut.setter
    def dateDebut(self, value: datetime):
        self._dateDebut = value

    @property
    def dateLivraison(self) -> datetime:
        return self._dateLivraison

    @dateLivraison.setter
    def dateLivraison(self, value: datetime):
        self._dateLivraison = value

    @property
    def prix(self) -> float:
        return self._prix

    @prix.setter
    def prix(self, value: float):
        self._prix = value

    @property
    def etat(self) -> Etat:
        return self._etat

    @etat.setter
    def etat(self, value: Etat):
        self._etat = value

File: src/model/Contracts.ts
enum Etat {
  Attente,
  EnCours,
  Fini,
  Retard,
  Rejected,
  Approved
}

export class Contracts {

  private _id !: number;
  private _email !: string;
  private _idService !: number;
  private _dateDebut !: Date;
  private _dateLivraison !: Date;
  private _prix !: number;
  private _etat !: Etat;


  constructor(id: number, email: string, idService: number, dateDebut: Date, dateLivraison: Date, prix: number, etat: Etat) {
    this._id = id;
    this._email = email;
    this._idService = idService;
    this._dateDebut = dateDebut;
    this._dateLivraison = dateLivraison;
    this._prix = prix;
    this._etat = etat;
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get email(): string {
    return this._email;
  }

  set email(value: string) {
    this._email = value;
  }

  get idService(): number {
    return this._idService;
  }

  set idService(value: number) {
    this._idService = value;
  }

  get dateDebut(): Date {
    return this._dateDebut;
  }

  set dateDebut(value: Date) {
    this._dateDebut = value;
  }

  get dateLivraison(): Date {
    return this._dateLivraison;
  }

  set dateLivraison(value: Date) {
    this._dateLivraison = value;
  }

  get prix(): number {
    return this._prix;
  }

  set prix(value: number) {
    this._prix = value;
  }

  get etat(): Etat {
    return this._etat;
  }

  set etat(value: Etat) {
    this._etat = value;
  }
}
